src/account/PasskeyAccount.sol

```diff
@@ -0,0 +1,260 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import { Base64Url } from "../utils/Base64Url.sol";
+import { IEntryPoint, UserOperation } from "@aa/interfaces/IEntryPoint.sol";
+import { IPasskeyAccount, Passkey } from "../interfaces/IPasskeyAccount.sol";
+import { P256Verifier } from "../utils/P256Verifier.sol";
+import { SimpleAccount } from "@aa/samples/SimpleAccount.sol";
+
+contract PasskeyAccount is SimpleAccount, IPasskeyAccount {
+    P256Verifier public immutable p256Verifier;
+    Passkey private passkey;
+
+    // The constructor is used only for the "implementation" and only sets immutable values.
+    // Mutable value slots for proxy accounts are set by the 'initialize' function.
+    constructor(IEntryPoint anEntryPoint) SimpleAccount(anEntryPoint) {
+        // Get a P256Verifier instance from the mainnet or Sepolia testnet.
+        // p256Verifier = P256Verifier(0xc2b78104907F722DABAc4C69f826a522B2754De4);
+        p256Verifier = new P256Verifier();
+    }
+
+    /**
+     * The initializer for the PassKeysAcount instance.
+     * @param credId the credentail id of the passkey.
+     * @param pubKeyX public key X val from a passkey that will have a full ownership and control of this account.
+     * @param pubKeyY public key X val from a passkey that will have a full ownership and control of this account.
+     */
+    function initialize(
+        string calldata credId,
+        uint256 pubKeyX,
+        uint256 pubKeyY
+    ) external initializer {
+        super._initialize(address(0));
+        _initPasskey(credId, pubKeyX, pubKeyY);
+    }
+
+    function updatePasskey(
+        string calldata credId,
+        uint256 pubKeyX,
+        uint256 pubKeyY
+    ) external virtual override {
+        require(msg.sender == address(this), "Only wallet can update passkeys");
+        require(pubKeyX != 0 && pubKeyY != 0, "Zero passkey is not allowed");
+        require(
+            passkey.pubKeyX != 0 && passkey.pubKeyY != 0,
+            "Passkey doesn't exist"
+        );
+        passkey.credIdHash = keccak256(abi.encodePacked(credId));
+        passkey.pubKeyX = pubKeyX;
+        passkey.pubKeyY = pubKeyY;
+        emit PasskeyUpdated(credId, passkey.pubKeyX, passkey.pubKeyY);
+    }
+
+    function validateSignature(
+        UserOperation calldata userOp,
+        bytes32 userOpHash // As Webauthn's challenge
+    ) external view returns (uint256) {
+        return _validateSignature(userOp, userOpHash);
+    }
```

Comment:
Q: Is this function helping us to test? Inheriting SimpleAccount we only need to override `_validateSignature`.

Reply:
Since `_validateSignature()` inherits from `SimpleAccount`, it is restricted to internal. To enable its use in Foundry Test (or potential future applications for advanced signature verification), an external `validateSignature()` method needs to be made accessible.